rfd/0225-teleport-extended-support.md

```diff
@@ -0,0 +1,146 @@
+require a significant change in our deprecation and compatibility philosophies.
+
+When upgrading to LTS N from LTS N-1, the control plane will first be upgraded to LTS N,
+and all agents and clients will still be on LTS N-1. Which means we need to ensure compatibility
+for whatever the time span is between two subsequent releases.
+
+In addition to the technical and philosophical challenges that an LTS release would introduce, it would
+also have an impact on the tools team and developer experience. We would need to support an additional
+number of release branches and build pipelines. 
+
+### Extended Support
+
+While customers might be asking for an LTS release explicitly, what they really want are two
+things: stability and extended support. We can achieve these without introducing a Teleport LTS
+release by changing our process slightly.
+
+We can reduce the frequency of Teleport major releases from every ~4 months to once a year and cut
+down the number of supported releases from N-2 to N-1. That would extend the support window for major
+releases from ~12 months to ~24 months at the expense of supporting fewer major versions. Reducing
+the number of versions also means it takes fewer upgrade cycles to get to the most recent version of Teleport.
+
+This satisfies customer desire for major releases to be supported for a longer period of time
+without impacting our current upgrade process or component compatibility guarantees. However, it
+comes at the expense of reducing the frequency at which we can make breaking changes.
+
+As dictated by semantic versioning, a major release should be reserved for making any backward-incompatible
+changes to the public API. We've often relied on this four month release cycle to quickly iterate
+on APIs that don't work or cause issues for one reason or another. With yearly major releases this
+becomes a less viable solution and a stronger emphasis must be put both on backward and forward compatibility. 
+
+Going forward, solutions to problems should attempt to preserve compatibility at all costs and
+rely less on all customers adhering to our upgrade procedure. Instead of relying on versions of
+clients, and that the Auth server will reject outdated clients we should put a stronger emphasis
+on negotiating compatibility. If clients and servers in our control can inform each other about
+what features they support in some manner then both can mutually agree on which features and APIs
+can be safely used.
+
+To address the stability concerns, all new features *MUST* only be added to the most recent major version.
+The previous major version goes into maintenance mode the moment a new major version is released.
+In other words, even though we will now support two major releases, N and N-1, new features *MUST* only
+be included in version N. N-1 will continue to receive bug fixes and security patches, but not new features.
```

Comment:
this doesn't necessarily pertain to the core of the extended support change, feel free to resolve this comment if you want to avoid derailing the conversation, but I would be so happy if we could start cutting minor release branches off of master and never have to backport another feature.

For example, when we release v19, we would cut `branch/v19.0`. We would only backport bugfixes to `branch/v19.0`, which would actually align with semver. Feature slated for 19.1 would be merged to master, and when we're ready to release 19.1 we would cut `branch/v19.1` from master. Optionally we do the test plan again for 19.1 to continue running through a test plan every 4 months. Once 19.1 is released we can basically retire `branch/v19.0` and only backport bugfixes to `branch/v19.1`

Reply:
I think we rely on master being the staging area for larger features to land in together in a release, so I don't know how much that'd be practical.